esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -3827,7 +3828,7 @@ void CWsWuFileHelper::createWUZAPFile(IEspContext& context, IConstWorkUnit* cwu,
     WsWuInfo winfo(context, cwu);
     createZAPWUXMLFile(winfo, inFileNamePrefixWithPath.str());
     createZAPWUGraphProgressFile(request.wuid.str(), inFileNamePrefixWithPath.str());
-    createZAPWUQueryAssociatedFiles(cwu, folderToZIP);
+    createZAPWUQueryAssociatedFiles(cwu, folderToZIP, postMortemFolder);
```

Comment:
this function is a little confusing - it's not obvious from it's name that it is doing 2 things, copying associated files, and retrieving a specific category of file type (postmortem).

In bare-metal - we have to copy files because they are remote and accessible only via dafilesrv, not directly avail to zip command.
But in containerized mode, we expect everything to be local (for these files to be located at localhost), we should try to avoid copying files unnecessarily. i.e. they could be fed to the command line directly.

So I think this should be reworked, so it gets all local files (which will be everything in cloud), and feeds that list to the zip command.
To be generic and cater for the non-local files in bare-metal, it can copy those to a temp folder (see other comment re. using getTempFilePath())
The zip command can be passed the list of all files, those it did not have to copy, and those in the temp dir. it did have to copy, but can exclude the temp. directory name (with zip -j or zip -D). NB: can use <tempdir>/* on command line.

Either the postmortem files can be done at same time (no directory structure), which I think is prob. best/simplest, or you could add a via a 2nd zip command, in order to preserve the directory structure, but we really want the whole path of the debug directory in the zip file anyway.

Reply:
> To be generic and cater for the non-local files in bare-metal, it can copy those to a temp folder (see other comment re. using getTempFilePath())

In bare-metal, the ip for local files may not be 'localhost'). Should I use IpAddress.isLocal to identify a local file?

Reply:
> The zip command can be passed the list of all files

@jakesmith does the list include the file names and paths for every postmortem files?

Reply:
>In bare-metal, the ip for local files may not be 'localhost'). Should I use IpAddress.isLocal to identify a local file?

Yes, no need to differentiate the code between bare-metal and containerized, in containerized they will likely all be isLocal, but the code can be common.

Reply:
>>The zip command can be passed the list of all files
>@jakesmith does the list include the file names and paths for every postmortem files?

you can list the postmortem directory + "/*" , but you don't really want to keep the whole directory structure (e.g. the /var/lib/HPCCSystems/..)
You could avoid that by adding all other files 1st, then changing to the parent of the postmortem directory, and using zip command again to add it.

Reply:
The postmortem directory contains a subfolder. The /* does not work for subfolders:
/var/lib/HPCCSystems/debug/hthor-w20220322-172110-8s27s/2022-03-22T17:21:15+00:00/info.err
/var/lib/HPCCSystems/debug/hthor-w20220322-172110-8s27s/2022-03-22T17:21:15+00:00/info.log
/var/lib/HPCCSystems/debug/hthor-w20220322-172110-8s27s/2022-03-22T17:21:15+00:00/postmortem.log.0
/var/lib/HPCCSystems/debug/hthor-w20220322-172110-8s27s/2022-03-22T17:21:15+00:00/etc/config/hthor.yaml

Reply:
You should be able to use zip -r in the parent directory of the postmortem and name it to get what we want - i.e. capture the structure of the post-mortem directory in the zip file, without the leading platform directories.

Reply:
> Either the postmortem files can be done at same time (no directory structure), which I think is prob. best/simplest, or you could add a via a 2nd zip command, in order to preserve the directory structure, but we really do not want the whole path of the debug directory in the zip file anyway.

In [cc19b0b](https://github.com/hpcc-systems/HPCC-Platform/pull/15895/commits/cc19b0b8fd62b1626997207eb7abf78463cc4926), I do add the postmortem files (no directory structure) into zipped ZAP report. Do we really want to capture the structure of the post-mortem directory?